### PR TITLE
[2.x] fix: Use JDK path, not JRE path

### PR DESCRIPTION
**Pull request by eed3si9n**
## Problem
There are a few places where javaHome or java path is set, using java.home system property. The problem is that it points to JRE, not JDK, so it would break on Java compilation etc.

## Solution
If the path ends with jre, go up one directory.

### CODE REVIEW

**Comment by jtjeferreira** [comment]
I need to have a closer look, but this assumes the JRE is "contained" in the JDK. Is that always true? Maybe this is just a setup problem and a better error needs to be returned stating that javaHome doest not point to a JDK/javac?

PS: how is this currently handled when SBT needs to compile mixed scala and java code and it needs to call javac?

**Comment by eed3si9n** [comment]
JAVA_HOME environment variable and PATH (we can sort of think of it as an env variable) are typically both correct, and point at JDK. when some piece of code uses the System Property `java.home` that's when we get into trouble because it points to a "JRE" version by spec. I am assuming that anyone running sbt uses JDK, and that JRE is a synthetic one within a JDK.

File: server-test/src/test/scala/testpkg/BuildServerTest.scala [hunk 1]
```diff
@@ -257,16 +256,17 @@ class BuildServerTest extends AbstractServerTest {
...
     IO.write(
       otherBuildFile,
       """
+        |def jdk: File = sbt.internal.Util.javaHome.toFile()
         |lazy val javaProj = project
         |  .in(file("java-proj"))
         |  .settings(
         |    javacOptions += "-Xlint:all",
-        |    javaHome := Some(file(System.getProperty("java.home")))
+        |    javaHome := Some(jdk)
```
Comment (eed3si9n) [inline server-test/src/test/scala/testpkg/BuildServerTest.scala:269]: Not sure how this has been working until João's File PR exposed this bug.

**Comment by eed3si9n** [comment]
@jtjeferreira See https://docs.oracle.com/javase/tutorial/essential/environment/sysprop.html

> "java.home" | Installation directory for Java Runtime Environment (JRE)

**Comment by Friendseeker** [comment]
Java never ceases to bring surprise.

Usually JDK and JRE installation directory are the same, but I guess there's always exception.

**Pull request merged**